src/interpolations.jl

```diff
@@ -16,6 +16,12 @@ const FloatColorant{T<:AbstractFloat} = Colorant{T}
 @inline _make_compatible(::Type{T}, fill::Number) where {T} = T(fill)
 
 Interpolations.tweight(A::AbstractArray{C}) where C<:Colorant{T} where T = T
+if all(map((FixedPoint, Normed)) do T
+        isempty(methodswith(AbstractArray{<:T}, Interpolations.tweight))
+    end)
+    # Reached when upstream package (e.g., FixedPoint) doesn't provide a specialization
+    Interpolations.tweight(A::AbstractArray{T}) where T<:FixedPoint = T
+end
```

Comment:
This is what I can come up with so far to minimize the danger of type piracy (and incremental compilation). Do you have any better ideas? @timholy

Reply:
If this is really an issue, I suspect we need to add a `Requires` dependency to Interpolations and put this in a `@require` block there. This solution seems likely to be fragile with respect to load order, and may not be fully correct wrt precompilation.

Reply:
> may not be fully correct wrt precompilation.

`Requires` will disable the precompilation of this line of code, and thus won't get any trouble, am I understanding it correctly?

Reply:
Correct. I'm also not completely clear on whether the precompiled module would retain this runtime check of the method tables; I think it wouldn't, it would just store the answer at the time when it happened to run. In contrast, by putting it into the `__init__` method it's guaranteed to be correct.